src/context/GlobalStateProvider.tsx

```diff
@@ -0,0 +1,49 @@
+import { useInterpret } from '@xstate/react';
+import React, { createContext } from 'react';
+import { Interpreter } from 'xstate';
+import {
+  notificationMachine,
+  NotificationsContext,
+  NotificationsEvent,
+} from '../state/global/notifications/notificationMachine';
+import {
+  LoginNavigationContext,
+  LoginNavigationEvent,
+  loginNavigationMachine,
+  LoginNavigationState,
+} from '../state/global/ui/loginNavigationMachine';
+import {
+  UserSegmentContext,
+  UserSegmentEvent,
+  userSegmentMachine,
+  UserSegmentState,
+} from '../state/global/ui/userSegmentMachine';
+
+interface GlobalStateContextProps {
+  ui: {
+    loginNavigationService: Interpreter<LoginNavigationContext, any, LoginNavigationEvent, LoginNavigationState>;
+    userSegmentService: Interpreter<UserSegmentContext, any, UserSegmentEvent, UserSegmentState>;
+  };
+  notificationsService: Interpreter<NotificationsContext, any, NotificationsEvent>;
```

Comment:
Isn't the notification service part of the ui node?

Reply:
No, the ui node is related only to things that contains UI state, like if drawer is open or not.